lib/verify.js

```diff
@@ -135,6 +160,34 @@ export async function verify({
     expired: undefined
   };
 
+  const isEnvelopedVP = presentation?.type ===
+    'EnvelopedVerifiablePresentation';
+  if(returnExchange || isEnvelopedVP) {
+    // fetch unenveloped presentation from exchange state
+    try {
+      const response = await zcapClient.read({url: exchangeId, capability});
+      const {data: {exchange}} = response;
+      if(returnExchange) {
+        result.exchange = exchange;
+        return result;
+      }
+      // use exchange state to update `credential`
+      if(isEnvelopedVP) {
+        // update `credential`
+        const {getCredentialFromExchange} = options;
+        credential = await getCredentialFromExchange({exchange, credential});
+        result.credential = credential;
+      }
```

Comment:
This seems like it should be updating the presentation, not a credential.

We need some tests here.